js/assessment.js

```diff
@@ -1,409 +1,397 @@
+      if (!data._isPartOfAssessment) return;
+
+      const related = _.pluck(getRelatedQuestions(data), '_id');
+
+      return related.indexOf(data._id) + 1;
+    });
+
+    Handlebars.registerHelper('questionCount', function getTotalQuestions() {
+      const data = this.view ? this.view.model.toJSON() : this;
+      if (!data._isPartOfAssessment) return;
+      return getRelatedQuestions(data).length;
+    });
+  },
+
+  // Public functions
+  register(assessmentModel) {
+    const state = assessmentModel.getState();
+    const assessmentId = state.id;
+    const pageId = state.pageId;
+
+    if (this._assessments._byPageId[pageId] === undefined) {
+      this._assessments._byPageId[pageId] = [];
+    }
 
-      this._restoreModelState(assessmentModel);
-      this._restoredCount++;
+    this._assessments._byPageId[pageId].push(assessmentModel);
 
-      Adapt.trigger('assessments:register', state, assessmentModel);
+    this._addToAssessmentIdMap(assessmentId, assessmentModel);
 
-      this._setPageProgress();
+    this._assessments.push(assessmentModel);
 
-      this._setupQuestionNumbering();
+    this._restoreModelState(assessmentModel);
+    this._restoredCount++;
 
-      if (this._restoredCount === this._assessments.length) {
-        // Since all assessments have been stored, broadcast an
-        // event which has the collated state.
-        Adapt.trigger('assessment:restored', this.getState());
-      }
-    },
+    Adapt.trigger('assessments:register', state, assessmentModel);
 
-    get: function(id) {
-      if (id === undefined) {
-        return this._assessments.slice(0);
-      } else {
-        return this._assessments._byAssessmentId[id];
-      }
-    },
+    this._setPageProgress();
 
-    saveState: function() {
+    this._setupQuestionNumbering();
 
-      this._saveStateModel = {};
-      for (var i = 0, assessmentModel; assessmentModel = this._assessments[i++];) {
-        var state = assessmentModel.getState();
-        this._saveStateModel[state.id] = Adapt.offlineStorage.serialize(assessmentModel.getSaveState());
-      }
+    if (this._restoredCount === this._assessments.length) {
+      // Since all assessments have been stored, broadcast an
+      // event which has the collated state.
+      Adapt.trigger('assessment:restored', this.getState());
+    }
+  },
 
-      Adapt.offlineStorage.set('a', this._saveStateModel);
-    },
+  get(id) {
+    if (id === undefined) {
+      return this._assessments.slice(0);
+    } else {
+      return this._assessments._byAssessmentId[id];
+    }
+  },
 
-    getConfig: function () {
-      var assessmentsConfig = Adapt.course.get('_assessment');
+  saveState() {
 
-      if (assessmentsConfig && assessmentsConfig._isDefaultsLoaded) {
-        return assessmentsConfig;
-      }
+    this._saveStateModel = {};
+    for (let i = 0; i < this._assessments.length; i++) {
+      const assessmentModel = this._assessments[i];
```

Comment:
`for...of` loops are nice in these situations:
```suggestion
    for (const assessmentModel of this._assessments) {
```